src/components/ModalComponent.tsx

```diff
@@ -38,144 +42,175 @@ const getSelectionCoordinates = () => {
+      window.addEventListener('keydown', escapeListener, true)
 
-  /** Dispatches a tutorial action that ends the tutorial. */
-  endTutorial = () => this.props.dispatch(tutorial({ value: false }))
-
-  render() {
-    const {
-      show,
-      id,
-      title,
-      arrow,
-      center,
-      opaque,
-      className,
-      style,
-      actions,
-      positionAtCursor,
-      hideModalActions,
-      top,
-      children,
-    } = this.props
-
-    if (!show) return null
-
-    const cursorCoords = getSelectionCoordinates()
-
-    /** Dispatches a closeModal action for the modal. */
-
-    // /** Dispatches a modalComplete action for the modal. */
-    // const complete = () => dispatch(modalComplete(id))
-
-    // /** Dispatches a tutorial action that ends the tutorial. */
-    // const endTutorial = () => dispatch(tutorial({ value: false }))
-
-    return (
-      <div
-        ref={this.ref}
-        style={Object.assign(
-          {},
-          style,
-          top && { top: 55 },
-          positionAtCursor && {
-            top: cursorCoords.y,
-            left: cursorCoords.x,
-          },
-        )}
-        className={
-          className +
-          ' ' +
-          classNames({
-            modal: true,
-            animate: true,
-            [`modal-${id}`]: true,
-            center,
-            opaque,
-          })
-        }
-      >
-        {!this.props.preventCloseOnEscape && (
-          <a className='upper-right popup-close-x text-small' onClick={this.close}>
-            ✕
-          </a>
-        )}
-        <div
-          className={classNames({
-            'modal-content': true,
-            ...(arrow && { [arrow]: arrow }),
-          })}
-        >
-          {title && <h1 className='modal-title'>{title}</h1>}
-          <div className='modal-text'>{children}</div>
-          {!hideModalActions && actions && (
-            <div className='modal-actions'>
-              {actions({
-                close: this.close,
-                complete: this.complete,
-              })}
-            </div>
+      return () => {
+        modalCleanup()
+        window.removeEventListener('keydown', escapeListener!, true)
+      }
+    }
+  }, [])
+
+  const {
+    show,
+    id,
+    title,
+    arrow,
+    center,
+    opaque,
+    className,
+    style,
+    actions,
+    positionAtCursor,
+    hideModalActions,
+    top,
+    children,
+  } = props
+
+  if (!show) return null
+
+  const cursorCoords = getSelectionCoordinates()
+
+  return (
+    <AnimatePresence>
+      {!closing && (
+        <ModalWrapper
+          ref={ref}
+          opaque={opaque}
+          center={center}
+          initial={{
+            opacity: 0,
+          }}
+          animate={{
+            opacity: 1,
+          }}
+          exit={{
+            opacity: 0,
+          }}
+          transition={{ duration: 0.5 }}
+          style={Object.assign(
```

Comment:
Let's replace `Object.assign` with the spread operator while we're here.

Reply:
Sure.

Reply:
TODO